### ISSUE TITLE
Develop a way to flexibly add parameters of length Ncomponents to support catalog specific data

### ISSUE THREAD

**Issue by bhazelton** [RadioAstronomySoftwareGroup/pyradiosky#126]
Catalogs often have extra information per component. Rather than having to add optional parameters by hand for each one, we should develop a mechanism to support this kind of thing (including round tripping through files). Maybe something like `extra_keywords` but for arrays of length Ncomponents.

**Comment by bhazelton** [comment]
Need a method that takes an Ncomponent length array, a name and an expected type and adds a new UVParameter to the object with the properties set up properly. Maybe "register_new_parameter" (is parameter overloaded?).

places to modify code for it:
- init?
- concat
- select
- write/read skyh5
- warn about dropping it in write text file

**Comment by steven-murray** [comment]
I think `register_new_parameter` is not the best, since I tend to think of parameters as single-valued things that affect other things, rather than _data_. Maybe, `add_component_attribute`?

**Comment by bhazelton** [comment]
To round-trip them through `skyh5` files we may want to write a list of these added parameters to some header field in the file.

**Comment by bhazelton** [comment]
We ran into a roadblock when we tried to implement this because in python, you cannot add a property to an instance, only to a class. So when we tried to add the property to the instance, we changed all future instances also, which is not what we wanted to do.

We are now thinking about using a recarray to store all of these extra Ncomponent length parameters. We'd like to make it so that adding them typically (always?) happens via a method but that accessing them is pretty easy.
